spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/support/TaskRepositoryFactoryBean.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.repository.support;
+
+import org.springframework.cloud.task.repository.TaskRepository;
+
+/**
+ * Automates the creation of a {@link SimpleTaskRepository} using DAO implementations
+ * which persist task execution data to a datastore.
+ *
+ * @author Glenn Renfro
+ */
+public interface TaskRepositoryFactoryBean {
```

Comment:
Do we really need this interface if we use Spring's `FactoryBean` interface (since generics address the typing)?